slack-request-verifier/src/main/java/com/hubspot/slack/client/request/verifier/SlackRequestVerifierFilter.java

```diff
@@ -0,0 +1,151 @@
+package com.hubspot.slack.client.request.verifier;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+
+import org.apache.commons.codec.binary.Hex;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
+import com.google.common.io.CharStreams;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+
+/**
+ * Check for detailed information https://api.slack.com/authentication/verifying-requests-from-slack
+ */
+public class SlackRequestVerifierFilter implements ContainerRequestFilter {
+
+  private static final String SLACK_TIMESTAMP_HEADER = "X-Slack-Request-Timestamp";
+  private static final String SLACK_SIGNATURE_HEADER = "X-Slack-Signature";
+  private static final String CONCATENATED_SIGN_TEMPLATE = "%s:%s:%s";
+  private static final String ENCODED_SIGNATURE_TEMPLATE = "%s=%s";
+  private static final String VERSION_NUMBER = "v0";
+
+  private final long requestExpirationTime;
+  private final String signingSecret;
+
+  public interface Factory {
+    SlackRequestVerifierFilter create(
+        @Assisted("requestExpirationTime") long requestExpirationTime,
+        @Assisted("signingSecret") String signingSecret
```

Comment:
IIRC, we don't have to specify `@Assisted` annotations on the factory params.

Reply:
We do so in our internal projects so I thought that it works like that.
I will try to remove them